apstools/devices/positioner_soft_done.py

```diff
@@ -144,7 +144,7 @@ def __init__(
         **kwargs,
     ):
 
-        if setpoint_pv == readback_pv:
+        if setpoint_pv == readback_pv and setpoint_pv != "":
```

Comment:
This additional logic causes a unit test to fail: https://github.com/BCDA-APS/apstools/blob/a3c3c92d8f6be4807024eb765ea69846d5774af6/apstools/devices/tests/test_positioner_soft_done.py#L45-L48

Is there a reason to allow both readback_pv and setpoint_pv to be empty strings?

Reply:
I came across this issue because I wanted to set `setpoint` and `readback` in the child class, so both `setpoint_pv` and `readback_pv` `kwargs` were `""`

Reply:
Aren't some additional checks necessary then so that PVPositionerSoftDone does not attempt to proceed with these empty strings AND an defined readback and/or setpoint Components?

See an example of what you describe in class [`apstools.devices.Eurotherm2216e`](https://github.com/BCDA-APS/apstools/blob/40a79e3ce6d70449e2979a193ed33380e1d504fa/apstools/devices/eurotherm_2216e.py#L32)

https://github.com/BCDA-APS/apstools/blob/40a79e3ce6d70449e2979a193ed33380e1d504fa/apstools/devices/eurotherm_2216e.py#L39-L46

and the constructor: https://github.com/BCDA-APS/apstools/blob/40a79e3ce6d70449e2979a193ed33380e1d504fa/apstools/devices/eurotherm_2216e.py#L79-L88

The unmodifed logic can handle this and prevent the error from empty strings with no Components.

Reply:
The [Eurotherm's unit test](https://github.com/BCDA-APS/apstools/blob/40a79e3ce6d70449e2979a193ed33380e1d504fa/apstools/devices/tests/test_eurotherm_2216e.py#L14) is trivial since the controller is not generally available to a unit test workflow.  https://github.com/BCDA-APS/apstools/blob/40a79e3ce6d70449e2979a193ed33380e1d504fa/apstools/devices/tests/test_eurotherm_2216e.py#L14-L31

However, it does test that no errors are generated when creating the object (including the setpoint and readback Components).  Connection with EPICS is not tested.

Reply:
Perhaps we should then modify the `PVPositionerSoftDone` `__init__` to the one below?

```
    def __init__(
        self,
        prefix="",
        *,
        readback_pv="ignoreRBV",
        setpoint_pv="ignore",
        tolerance=None,
        update_target=True,
        **kwargs,
    ):
```

Reply:
And modify the logic back to `if setpoint_pv == readback_pv:`

Reply:
That seems reasonable since a different signal class is desired.  Same decision as the Eurotherm controller support.